src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java

```diff
@@ -39,8 +39,14 @@ public ModificationGroupEntity(UUID uuid) {
     }
 
     public void addModification(ModificationEntity modification) {
-        modifications.add(modification);
-        modification.setGroup(this);
+        if (modifications.isEmpty()) {
+            List<ModificationEntity> newList = new ArrayList<>();
+            newList.add(modification);
+            setModifications(newList);
```

Comment:
`setModifications(List.of(modification)) `?

Reply:
This was my first shot. But I we pass an unmutable collection like List(), we got an exception when we try to add a second element to the modifications list.

Reply:
Arrays.asList

Reply:
Arrays.asList also returns "a fixed-size list"

Reply:
same issue with:
 List<ModificationEntity> newList = Arrays.asList(modification);

Reply:
sonar doesnt like it ;-)
_Double Brace Initialization should not be used_